src/exceptions/invalidargumentexception.h

```diff
@@ -0,0 +1,16 @@
+#ifndef INSTANT_TRANSLATOR_INVALIDARGUMENTEXCEPTION_H
+#define INSTANT_TRANSLATOR_INVALIDARGUMENTEXCEPTION_H
+
+#include <QException>
+
+/**
+ * Invalid argument exception.
+ */
+class InvalidArgumentException : public QException {
+public:
+    void raise() const { throw *this; }
+
+    InvalidArgumentException *clone() const { return new InvalidArgumentException(*this); }
```

Comment:
`return new...` should be in new line as well as ending bracket `}`.

Reply:
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-inline

Reply:
This is for `inline` functions, while this one is not an `inline` function, its a stateful method.